### PR TITLE
tests/bitarithm_timings: Fixed bugs

### PR DESCRIPTION
**Pull request by maribu**
### Contribution description

Fixed bugs in tests/bitarithm_timings:

- Prevent `bitarithm_lsb()` from being called with 0, as it loops forever then
- Use C11 atomics for communication with IRQ context, instead of abusing `volatile` for stuff it is not meant to be used

### Testing procedure

The test should now pass on ATmegas

### Issues/PRs references

First item in https://github.com/RIOT-OS/RIOT/issues/12651

### CODE REVIEW

**Comment by maribu** [comment]
I think nobody started reviewing, so I squash the missing include in directly.

File: tests/bitarithm_timings/main.c [hunk 1]
```diff
@@ -38,26 +40,29 @@
...
-    xtimer_t xtimer;
-    xtimer.callback = callback;
-    xtimer.arg = (void *) &done;
-
+    xtimer_t xtimer = { .callback = callback };
     xtimer_set(&xtimer, TIMEOUT);
 
     do {
-        if (i++ == -1u) {
+        if (++i == UINT_MAX) {
```
Comment (aabadie) [inline tests/bitarithm_timings/main.c]: I just checked and only this change is necessary. The problem is that the change is provided in the same commit as the use C11 atomic. Just moving the latter in a separate commit would be better for me.

Comment (maribu) [reply tests/bitarithm_timings/main.c]: Done

**Comment by maribu** [comment]
Thanks for the review!
